sample_project/Assets/SampleViewer/Samples/HitTest/ArcGISRaycast.cs

```diff
@@ -5,30 +5,111 @@
 //
 
 using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
 using Esri.ArcGISMapsSDK.Components;
 using Esri.GameEngine.Geometry;
+using Newtonsoft.Json.Linq;
 using TMPro;
 using UnityEngine.InputSystem;
 using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.Networking;
 using UnityEngine.UI;
 
 public class ArcGISRaycast : MonoBehaviour
 {
-    [SerializeField] private InputAction inputAction;
-    private const float offSet = 200f;
-
     public ArcGISMapComponent arcGISMapComponent;
     public ArcGISCameraComponent arcGISCamera;
-    public Canvas canvas;
-    public TextMeshProUGUI featureText;
+    private int featureId;
     private InputActions inputActions;
     private bool isLeftShiftPressed;
+    private JToken[] jFeatures;
+    [SerializeField] private TextMeshProUGUI locationText;
+    [SerializeField] private GameObject markerGO;
+    private List<String> outfields = new List<String>{"AREA_SQ_FT", "DISTRICT", "Height", "SUBDISTRIC", "ZONE_"};
+    private List<String> properties = new List<String>{"Area", "District", "Height", "Sub District", "Zone"};
+    [SerializeField] private TMP_Dropdown scrollView;
+    [SerializeField] private bool supressWarnings;
+    [SerializeField] private Image warningImage;
+    private string weblink;
+
+    [SerializeField] private TextMeshProUGUI property;
 
     private void Awake()
     {
         inputActions = new InputActions();
     }
 
+    private void CreateLink(string objectID)
+    {
+        weblink = "https://services.arcgis.com/P3ePLMYs2RVChkJx/ArcGIS/rest/services/Buildings_Boston_USA/FeatureServer/0/query?f=geojson&where=1=1&objectids=" + objectID + "&outfields=*";
+        StartCoroutine(GetFeatures(objectID));
+    }
+    
+    private IEnumerator GetFeatures(string objectID)
+    {
+        // To learn more about the Feature Layer rest API and all the things that are possible checkout
+        // https://developers.arcgis.com/rest/services-reference/enterprise/query-feature-service-layer-.htm
+
+        UnityWebRequest Request = UnityWebRequest.Get(weblink);
+        yield return Request.SendWebRequest();
+
+        if (Request.result != UnityWebRequest.Result.Success)
+        {
+            Debug.Log(Request.error);
+        }
+        else
+        {
+            if (scrollView.value < 0)
```

Comment:
```suggestion
        return;
        }

        if (scrollView.value < 0)
```

Reply:
Can't do this in a CoRoutine. Plus it wouldn't work because I still need to run the code in that if Statement if Scrollview.value < 0

Reply:
`yield break;`
What do you mean you still need to run the code? Anything in the else wont run if there is an error

Reply:
`                property.text = "Feature ID: " + objectID;
                warningImage.enabled = true;

                if (!supressWarnings)
                {
                    Debug.LogWarning("Please select a value to get from the dropdown");
                }`
                
That is the intention. This needs to be run if the scroll view is -1/below 0. It tells the user they don't have a property selected. They then go select a property and it re-runs this code. This is by design.